### ISSUE TITLE
Try Advanced Models like XGBoost / LightGBM

### ISSUE THREAD

**Issue by nupurmadaan04** [nupurmadaan04/AI-agriculture-yield-production#6]
Experiment with tree-based boosting models known for high performance:

Train using XGBoost and LightGBM
Compare their performance with the existing model
Tune their hyperparameters to maximize accuracy

📌 Goal: Achieve better accuracy with optimized boosting algorithms.

**Comment by almishhoda04** [comment]
hi @nupurmadaan04 i am a passionate 3rd year btech aiml student and also a first time contributor in Gssoc . I can experiment xgboost and lightgbm on the existitng rice_data_outlier_removed.csv file to generate results, compare and tune hyper parameters. 

I would love to contribute if you assign me this .

**Comment by nupurmadaan04** [comment]
@almishhoda04 Hi dear , This side Nupur and am glad to see your enthusiasm towards the project . Right now training and testing accuracy of the data is 99% and 95% so improve test accuracy. For any doubts feel free to contact me here / on discord / linkedin.
Am assigning this issue to you

**Comment by nupurmadaan04** [comment]
@vijyush this is assigned issue , you can look after for another issue or create it.
